docs/src/index.md

```diff
@@ -525,11 +525,165 @@ To reference columns with more complicated expressions, you must wrap column ref
+In operations, it is also allowed to use `AsTable(cols)` to work with
+multiple columns at once, where the columns are grouped together in a
+`NamedTuple`. When `AsTable(cols)` appears in a operation, no
+other columns may be referenced in the block.
 
-### Multi-argument selectors 
+`AsTable` on the right-hand side also allows the use of the special
+column selectors `Not`, `Between`, and regular expressions, as well
+as working with lists of variables programmatically. 
+
+For example, consider a collection of column names `vars`, such that
+
+```
+df = DataFrame(a = [11, 14], b = [17, 10], c = [12, 5]);
+vars = ["a", "b"];
+```
+
+To make a new column which is the sum of `vars`, write
+
+```
+julia> @rtransform df :y = sum(AsTable(vars))
+2×4 DataFrame
+ Row │ a      b      c      y     
+     │ Int64  Int64  Int64  Int64 
+─────┼────────────────────────────
+   1 │    11     17     12     28
+   2 │    14     10      5     24
+```
+
+Of course, you can also use `AsTable` on the right-hand side using `Symbol`s as column selectors
+
+```
+julia> @rtransform df :y = sum(AsTable([:a, :b]))
+2×4 DataFrame
+ Row │ a      b      c      y     
+     │ Int64  Int64  Int64  Int64 
+─────┼────────────────────────────
+   1 │    11     17     12     28
+   2 │    14     10      5     24
+```
+
+`AsTable` on the right-hand side also allows operations which can use the names of the variables. 
+
+```
+julia> function fun_with_new_name(x::NamedTuple)
+           nms = string.(propertynames(x))
+           new_name = join(nms, "_") * "_sum"
+           s = sum(x)
+           (; Symbol(new_name) => s)
+       end
+
+julia> @rtransform df $AsTable = fun_with_new_name(AsTable([:a, :b]))
+2×4 DataFrame
+ Row │ a      b      c      a_b_sum 
+     │ Int64  Int64  Int64  Int64   
+─────┼──────────────────────────────
+   1 │    11     17     12       28
+   2 │    14     10      5       24
+```
+
+To subset all rows where the sum is greater than `25`, write
+
+```
+julia> @rsubset df sum(AsTable(vars)) > 25
+1×3 DataFrame
+ Row │ a      b      c     
+     │ Int64  Int64  Int64 
+─────┼─────────────────────
+   1 │    11     17     12
+```
+
+To understand the how this works, recall that DataFrames.jl allows for
+`AsTable(cols)` to be a `source` in a `source => fun => dest` mini-language
+expression. As a consequence, the transformation call
+
+```
+:y = f(AsTable(cols)) 
+```
+
+becomes
+
+```
+AsTable(cols) => f => :y
+```
+
+Note that DataFrames does *not* allow `source => fun => dest` commands 
+to be of the form 
+
+```
+[AsTable(cols), :x] => f => :y
+```
+
+As a consequence, DataFramesMeta.jl does not allow any other column selectors to appear 
+inside the expression. The command
+
+```
+:y = sum(AsTable(cols)) + :d
+```
+
+will fail. 
+
+Finally, note that everyting inside `AsTable` is escaped by default.
+There is no ned to use `$` inside `AsTable` on the right-hand side.
+For example
+
+```
+:y = first(AsTable("a"))
```

Comment:
why is this an example of the behavior you describe? I guess you wanted `:a` not `"a"` - right?

Reply:
No this is correct. Usually when working with `"a"`, you would need to write `$"a"`. I guess I could make it a variable you have to interpolate.